# HG changeset patch
# User Rik <rik@octave.org>
# Date 1489678191 25200
#      Thu Mar 16 08:29:51 2017 -0700
# Node ID aed81b364903be7c0c20d53ee4a0e9480fadd49d
# Parent  953cb077757c8c084f2a2bd6761ebe516cab560c
Correct 350X slowdown in eps (bug #50561).

* data.cc (Feps): Move assignment of intermediate variable epsval to retval
from inside for loop to single execution after for loop.

diff --git a/libinterp/corefcn/data.cc b/libinterp/corefcn/data.cc
--- a/libinterp/corefcn/data.cc
+++ b/libinterp/corefcn/data.cc
@@ -4529,19 +4529,19 @@ type and may be either @qcode{"double"} 
                 epsval(i) = pow (2.0, -1074e0);
               else
                 {
                   int expon;
                   octave::math::frexp (val, &expon);
                   epsval(i) = std::pow (2.0,
                                         static_cast<double> (expon - 53));
                 }
-
-              retval = epsval;
             }
+
+          retval = epsval;
         }
     }
   else
     retval = fill_matrix (args, std::numeric_limits<double>::epsilon (),
                           std::numeric_limits<float>::epsilon (), "eps");
 
   return retval;
 }
